rcl/test/rcl/test_publisher.cpp

```diff
@@ -490,3 +535,294 @@ TEST_F(CLASSNAME(TestPublisherFixture, RMW_IMPLEMENTATION), test_invalid_publish
     rcl_publish_serialized_message(nullptr, &serialized_msg, nullptr));
   rcl_reset_error();
 }
+
+// Mocking rmw_publisher_count_matched_subscriptions to make
+// rcl_publisher_get_subscription_count fail
+TEST_F(CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION), test_mock_publisher_count) {
+  auto mock = mocking_utils::patch(
+    "lib:rcl", rmw_publisher_count_matched_subscriptions, [](auto...) {return RMW_RET_BAD_ALLOC;});
+
+  // Now normal usage of the function rcl_publisher_get_subscription_count returning
+  // unexpected RMW_RET_BAD_ALLOC
+  size_t count_size;
+  EXPECT_EQ(
+    RCL_RET_BAD_ALLOC, rcl_publisher_get_subscription_count(&publisher, &count_size));
+  rcl_reset_error();
+}
+
+// Mocking rmw_publisher_assert_liveliness to make
+// rcl_publisher_assert_liveliness fail
+TEST_F(CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION), test_mock_assert_liveliness) {
+  auto mock = mocking_utils::patch(
+    "lib:rcl", rmw_publisher_assert_liveliness, [](auto...) {return RMW_RET_ERROR;});
+
+  // Now normal usage of the function rcl_publisher_assert_liveliness returning
+  // unexpected RMW_RET_ERROR
+  EXPECT_EQ(
+    RCL_RET_ERROR, rcl_publisher_assert_liveliness(&publisher));
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+}
+
+// Mocking rmw_publish to make rcl_publish fail
+TEST_F(CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION), test_mock_publish) {
+  auto mock = mocking_utils::patch(
+    "lib:rcl", rmw_publish, [](auto...) {return RMW_RET_ERROR;});
+
+  // Test normal usage of the function rcl_publish returning unexpected RMW_RET_ERROR
+  test_msgs__msg__BasicTypes msg;
+  test_msgs__msg__BasicTypes__init(&msg);
+  msg.int64_value = 42;
+  ret = rcl_publish(&publisher, &msg, nullptr);
+  test_msgs__msg__BasicTypes__fini(&msg);
+  EXPECT_EQ(RCL_RET_ERROR, ret) << rcl_get_error_string().str;
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+}
+
+// Mocking rmw_publish_serialized_message to make rcl_publish_serialized_message fail
+TEST_F(
+  CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION), test_mock_publish_serialized_message)
+{
+  rcl_serialized_message_t serialized_msg = rmw_get_zero_initialized_serialized_message();
+  size_t initial_capacity_ser = 0u;
```

Comment:
@Blast545 nit: is `initial_capacity_ser` an incomplete name?

Reply:
Renamed with c7de0f1